qemu/tests/qemu_guest_agent.py

```diff
@@ -390,6 +393,250 @@ def gagent_check_set_vcpus(self, test, params, env):
             raise error.TestFail("the vcpu status is not changed as expected")
 
     @error.context_aware
+    def gagent_check_get_time(self, test, params, env):
+        """
+        Execute "guest-get-time" command to guest agent
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        os_type = params["os_type"]
+        error.context("get the time of the guest", logging.info)
+        nanoseconds_time = self.gagent.get_time()
+        if nanoseconds_time is False:
+            raise error.TestNAError("the  guest-get-time cmd is not supported")
+        error.context("the time get by guest-get-time is '%d' "
+                      % nanoseconds_time, logging.info)
+        guest_time = utils_misc.get_guest_os_datetime(self.vm, os_type)
+        if not guest_time:
+            raise error.TestError("can't get the guest time for contrast")
+        delta = (guest_time - nanoseconds_time / 1000000000)
+        if delta > 3:
+            raise error.TestFail("the time get by guest agent is not the same "
+                                 "with that by time check cmd inside guest")
```

Comment:
May be add a 'abs' operation is better , but I think it is just a in case operation, because It is extremely not possible that guest_time is less than nanoseconds_time.
I think that  "3s" or "1s" makes no difference for this test case. What I am considering is that if the host being in a high pressure , may be it need more time to get "guest_time", so the delta is set larger.
I will add the time info into log, that's a good idea, thanks.